hpobench/benchmarks/mo/cnn_benchmark.py

```diff
@@ -0,0 +1,535 @@
+            layers.append(nn.ReLU())
+            if config['batch_norm']:
+                layers.append(nn.BatchNorm2d(out_ch))
+            layers.append(nn.MaxPool2d(kernel_size=2, stride=2))
+            inp_ch = out_ch
+
+        self.conv_layers = nn.Sequential(*layers)
+        self.pooling = nn.AdaptiveAvgPool2d(1) if config['global_avg_pooling'] else nn.Identity()
+        self.output_size = num_classes
+
+        self.fc_layers = nn.ModuleList()
+
+        inp_n = self._get_conv_output(input_shape)
+
+        layers = [nn.Flatten()]
+        for i in range(config['n_fc_layers']):
+            out_n = config['fc_layer_{}'.format(i)]
+
+            layers.append(nn.Linear(inp_n, out_n))
+            layers.append(nn.ReLU())
+
+            inp_n = out_n
+
+        layers.append(nn.Linear(inp_n, num_classes))
+        self.fc_layers = nn.Sequential(*layers)
+
+        self.time_train = 0
+
+    # generate input sample and forward to get shape
+    def _get_conv_output(self, shape):
+        bs = 1
+        input = torch.autograd.Variable(torch.rand(bs, *shape))
+        output_feat = self.conv_layers(input)
+        output_feat = self.pooling(output_feat)
+        n_size = output_feat.data.view(bs, -1).size(1)
+        return n_size
+
+    def forward(self, x):
+        x = self.conv_layers(x)
+        x = self.pooling(x)
+        x = self.fc_layers(x)
+        return x
+
+    def train_fn(self, optimizer, criterion, loader, device):
+        """
+        Training method
+        :param optimizer: optimization algorithm
+        :param criterion: loss function
+        :param loader: data loader for either training or testing set
+        :param device: torch device
+        :return: accuracy on the data
+        """
+        accuracy = AccuracyTop1()
+        self.train()
+
+        for images, labels in loader:
+            images = images.to(device)
+            labels = labels.to(device)
+
+            # Step
+            optimizer.zero_grad()
+            logits = self(images)
+
+            loss = criterion(logits, labels.argmax(-1))
+            loss.backward()
+            optimizer.step()
+
+            acc = accuracy(labels, logits)
+
+        return acc
+
+    def eval_fn(self, loader, device):
+        """
+        Evaluation method
+        :param loader: data loader for either training or testing set
+        :param device: torch device
+        :param train: boolean to indicate if training or test set is used
+        :return: accuracy on the data
+        """
+        accuracy = AccuracyTop1()
+        self.eval()
+
+        with torch.no_grad():  # no gradient needed
+            for images, labels in loader:
+                images = images.to(device)
+                labels = labels.to(device)
+
+                outputs = self(images)
+                acc = accuracy(labels, outputs)
+
+        return acc
+
+
+class CNNBenchmark(AbstractBenchmark):
```

Comment:
After rebasing to the new state, we can use the MO interface

```suggestion
class CNNBenchmark(AbstractMultiObjectiveBenchmark):
```